app/src/main/java/com/lagradost/cloudstream3/ui/result/ResultFragmentTv.kt

```diff
@@ -646,15 +646,14 @@ class ResultFragmentTv : Fragment() {
         }
 
         observeNullable(viewModel.movie) { data ->
-            if (data == null) return@observeNullable
+            if (data == null || resumeMode) {
```

Comment:
observe does not have a determined order, so I dont like this. It just sounds like you are making the bug less likely

Reply:
That's why i am checking the resume this way, i didn't find another way to do this.
I am open for suggestions though

Reply:
I would move all the visibility logic into resumeWatching and make this function act as a setOnClickListener, then do the nextFocusRightId in the loadpage section. So this makes it so:

Page fills all the metadata about the thing
Movie adds the listener
Resume watching handles the visibility

But if this is too much the viewmodel should be simplified, as all the parts should be independent and support out of order eval

Reply:
What you can do is making a dummy header view, that way you can have an "X AND Y" visibility, this is very common in cs3 for these things

Reply:
> I would move all the visibility logic into resumeWatching and make this function act as a setOnClickListener, then do the nextFocusRightId in the loadpage section. So this makes it so:
> 
> Page fills all the metadata about the thing
> Movie adds the listener
> Resume watching handles the visibility
> 
> But if this is too much the viewmodel should be simplified, as all the parts should be independent and support out of order eval
> 
> 

I got that, will rework based on that

Reply:
> What you can do is making a dummy header view, that way you can have an "X AND Y" visibility, this is very common in cs3 for these things

I don't understand that bro 😂 please elaborate or guide me to a code in CS i can read to get the idea as an example

Reply:
> I don't understand that bro 😂 please elaborate or guide me to a code in CS i can read to get the idea as an example


Well the idea is to have a view that holds many items, see for example apply_btt_holder https://github.com/recloudstream/cloudstream/blob/81df68e1370e5c08169f07e1f69979691dad0ba6/app/src/main/res/layout/bottom_selection_dialog.xml#L36

Then you can change the visibility of the holder https://github.com/recloudstream/cloudstream/blob/81df68e1370e5c08169f07e1f69979691dad0ba6/app/src/main/java/com/lagradost/cloudstream3/utils/SingleSelectionHelper.kt#L116

But the neat thing is that it has a separate visibility from the view itself. So in this case if I wanted to toggle the visibility of apply by two bools, lets say realShowApply AND showApply then I can set the headers visibility to realShowApply and the apply buttons visibility to showApply. That way both the buttons (cancel and apply) are turned on if realShowApply is true, but the apply buttons is only visible if both realShowApply is true and showApply is true.

In your case I would group all buttons visible in one mode in one header view and turn that view on or off, but this will still keep the visibility state of each individual button.

Reply:
I handeled most of the visibility in ResumeWatching, and in Movie I check for Resume button visibility to toggle Play button visibility